pkg/authenticator/helmsecret.go

```diff
@@ -0,0 +1,33 @@
+package authenticator
+
+import (
+	"os"
+)
+
+type helmsecret struct {
+}
+
+var _ Authenticator = (*helmsecret)(nil)
+
+func NewHelmSecret() *helmsecret {
+	return &helmsecret{}
+}
+
+func (s *helmsecret) GetAuthFileName() (string, error) {
+	// Check if Helm registry config is set
+	helmconfig := os.Getenv("HELM_REGISTRY_CONFIG")
+	if helmconfig != "" {
+		// Use HELM_REGISTRY_CONFIG
+		return helmconfig, nil
+	}
+
+	// Check secret mount
```

Comment:
How about it just returns nil if HELM_REGISTRY_CONFIG is not set. This /secrets is a pod things, I'm not entirely sure it is correct and gosec doesn't like this.

Reply:
Well we need to pass in this authfile for the helm driver if it does exist. When this is running in a pod it would be there if its set and then it should be pass to the helm driver.

Reply:
The pod should set HELM_REGISTRY_CONFIG

Reply:
s/pod/deployment in the helm chart. The thing that sets where it is should set where it can be found.

Reply:
I agree with Terry here. This code should know as little as possible about how k8s works, and the fact that secrets are usually put in some specific place is 100% a k8s concern. This code should just look for the auth where the env var tells it.